mm2src/mm2_main/src/lp_swap/taker_swap.rs

```diff
@@ -1244,11 +1308,26 @@ impl TakerSwap {
 
         Ok((Some(TakerSwapCommand::WaitForTakerPaymentSpend), vec![
             TakerSwapEvent::TakerPaymentSent(tx_ident),
+            TakerSwapEvent::WatcherMessageSent(preimage_hex),
```

Comment:
But the watcher message is not sent if `preimage_fut.compat().await` returns an error